send_stream.go

```diff
@@ -421,6 +421,16 @@ func (s *sendStream) cancelWriteImpl(errorCode qerr.StreamErrorCode, remote bool
 	s.mutex.Lock()
 	if !remote {
 		s.cancellationFlagged = true
+		if s.cancelWriteErr != nil {
+			s.mutex.Unlock()
+			// The user has called CancelWrite. If the previous cancellation was
+			// because of a STOP_SENDING, we don't need to flag the error to the
+			// user any more.
+			if s.isNewlyCompleted() {
```

Comment:
This must be called while holding the mutex. I know it's a bit tricky. That's why you often see this pattern here:
```go
completed := s.isNewlyCompleted()
s.mutex.Unlock()
if completed {
    s.sender.onStreamCompleted(s.streamID)
}
```

Reply:
Done in 1f96f4f1